File: lib/hex_to_decimal.rb
def letter_values
  letter_values = {
    'A' => 10,
	  'B' => 11,
	  'C' => 12,
	  'D' => 13,
	  'E' => 14,
	  'F' => 15
  }
end

def get_value(item)
	if letter_values.has_key?(item.upcase)
		letter_values[item.upcase]
	else
		item.to_i
	end
end

def hex_to_decimal(hex_string)
	hex_string.reverse!
	hex_values = []

	hex_string.each_char { |char| hex_values << get_value(char) }
  dec_values = hex_values.map.with_index { |val, i| val * (16 ** i) }
	dec_values.reduce {|total, num| total += num}
end

File: lib/hex_to_decimal.js
var letterValues = {
  'A': 10,
  'B': 11,
  'C': 12,
  'D': 13,
  'E': 14,
  'F': 15,
}

var getHexValue = function(string) {
  if(letterValues[string] !== undefined) {
    return letterValues[string];
  } else {
    return parseInt(string, 10);
  }
}

var hexToDecimal = function(hex) {
  var placeValues = [];
  for(var i = 1; i <= hex.length; i++) {
    placeValues.push(getHexValue(hex[hex.length - i]) * Math.pow(16, i - 1))
  }
  var total = placeValues.reduce(function(total, each){
    return total + each;
  }, 0 );
  return total;
}

console.log(hexToDecimal('9')) // 9
console.log(hexToDecimal('E')) // 15
console.log(hexToDecimal('5A')) // 90
console.log(hexToDecimal('CC')) //204
